pygments/lexers/arturo.py

```diff
@@ -0,0 +1,312 @@
+
+        lexer = None
+        if self.handle_annotateds:
+            try:
+                lexer = get_lexer_by_name( match.group(3).strip() )
+            except ClassNotFound:
+                pass
+        code = match.group(5)
+
+        if lexer is None:
+            yield match.group(5), String, code
+        else:
+            yield from do_insertions([], lexer.get_tokens_unprocessed(code))
+
+        yield match.start(6), String.Double, match.group(6)
+
+
+
+    tokens = {
+        'root': [
+            include('comments'),
+            include('constants'),
+            include('builtin_functions'),
+            include('operators'),
+            (r'.', Name),
+        ],
+
+        'comments': [
+            (r'(;)(.*)$', Comment.Single),
+            include('shebang'),
+        ],
+            'shebang': [
+                (r'^((\s#!)|(#!))(.*)$',
+                        Comment.Hashbang)
+            ],
+
+        'operators': [
+            include('punctuation'),
+            include('sugar'),
+
+            (r'<:|:>|:<|:>'
+             r'|-:|:-'
+             r'|ø|∞'
+             r'|\@|\#|\$|\%|\&|\_|\!|\!\!'
+             r'|\+|\-|\*|\~|\=|\>|\<'
+             r'|==>|<=>|<==>'
+             r'|=>>|<<=>>|<<==>>'
+             r'|-->|<->|<-->'
+             r'|<\\|\<|\>'
+             r'|\=\||\|\='
+             r'|\./|\^|\.\.|\.'
+             r'|\\',
+            Operator)
+
+        ],
+            'punctuation': [
+                (r'[()[\],]',   Punctuation),
+            ],
+            'sugar':[
+                (r'->'
+                 r'|=>'
+                 r'|\|'
+                 r'|::',
+                Name.Decorator),
+            ],
+
+        'constants': [
+            include('boolean'),
+            include('character'),
+            include('color'),
+            include('float'),
+            include('integer'),
+            include('label'),
+            include('literal'),
+            include('strings'),
+            include('type'),
+        ],
+
+            'boolean': [
+                (words(('false', 'true', 'maybe'), suffix=r'\b'),
+                    Name.Constant)
+            ],
+            'character': [
+                (r'`.`', String.Char)
+            ],
+            'color': [
+                (r'#\w+',
+                        Name.Constant)
+            ],
+            'float': [
+                (r'[0-9]+\.[0-9]+',
+                        Number.Float)
+            ],
+            'integer': [
+                (r'[0-9]+',
+                        Number.Integer)
+            ],
+            'label': [
+                (r'\w+\b\??:',
+                        Name.Label)
+            ],
+            'literal': [
+                (r'\'(?:\w+\b\??:?)',
+                        Keyword.Declaration)
+            ],
+            'type': [
+                (r'\:\w+',
+                        Keyword.Type)
+            ],
+            'attributes': [
```

Comment:
this one is not used (another danger of using too many trivial states...)

Reply:
- [x] [remove additional state](https://github.com/pygments/pygments/pull/2259/commits/ddb7923d1aa888f2bce3577bc44f744ad4f55c32)
  - I tried to just remove `constants` state, but I think the commit bellow is better
- [x] [remove the high amount of trivial states](https://github.com/pygments/pygments/pull/2259/commits/cb6a6b7595b15e8971794adc701e8eea1730198b)
  - I just removed that amount of states, put on `constants`, and added side notes. I think it's better.

Reply:
There's still a few of them (punctuation, sugar, builtin-predicate-functions, maybe more...)

Reply:
- [x] [remove trivial states](https://github.com/pygments/pygments/pull/2259/commits/fab78186a488bb081a4e8e0b6db7fc1866f3ff98)

Reply:
@birkenfeld, do you think I need to remove `operators` state?

Reply:
"need to" is too strong, but in general states that are only used as an include once are not technically needed.

Reply:
Sure!